redirector.py

```diff
@@ -93,6 +93,11 @@ def update_stats(stats):
             if host in stats:
                 stats[host] = data
 
+def fetch_api_spawn(host):
+    http_client = HTTPClient()
```

Comment:
I would use an `AsyncHTTPClient()` here, so it doesn't block the main event loop. After that, you'll need...